precompiles/types/coin_test.go

```diff
@@ -20,11 +20,33 @@ func Test_createCoinSet(t *testing.T) {
 	tokenDenom := ZRC20ToCosmosDenom(tokenAddr)
 	amount := big.NewInt(100)
 
-	coinSet, err := CreateCoinSet(tokenAddr, amount)
+	coinSet, err := CreateZRC20CoinSet(tokenAddr, amount)
 	require.NoError(t, err, "createCoinSet should not return an error")
 	require.NotNil(t, coinSet, "coinSet should not be nil")
 
 	coin := coinSet[0]
 	require.Equal(t, tokenDenom, coin.Denom, "coin denom should be %s, got %s", tokenDenom, coin.Denom)
 	require.Equal(t, amount, coin.Amount.BigInt(), "coin amount should be %s, got %s", amount, coin.Amount.BigInt())
 }
+
+func Test_CoinIsZRC20(t *testing.T) {
+	test := []struct {
+		denom    string
+		expected bool
+	}{
+		{"", false}, // Empty string.
+		{"zrc20/", false}, // Missing address.
+		{"zrc20/0x514910771af9ca656af840dff83e8264ecf986ca", true}, // Valid ZRC20 address.
+		{"zrc20/0xCa14007Eff0dB1f8135f4C25B34De49AB0d42766", true}, // Valid ZRC20 address.
+		{"zrc200xabcdef", false}, // Malformed prefix.
+		{"foo/0x0123456789", false}, // Invalid prefix.
+		{"ZRC20/0x0123456789abcdef", false}, // Invalid prefix.
```

Comment:
```suggestion
		{"ZRC20/0x0123456789abcdef", false}, // Invalid prefix.
		{"zrc20/0x0123456789abcdef", false,   // valid prefix invalid address
```

Reply:
Added in 2b130fdc0f29a4d958d599d633c42f3987a6e7b5